### PR TITLE
feat(ui): Install clerk/elements

### PR DESCRIPTION
**Pull request by alexcarpenter**
## Description

- Installs `@clerk/elements` within UI package
- Installs `@clerk/nextjs` within theme-builder app
- Adds exports configuration to enable clean imports instead linking to dist directory
- Roughs in Elements components within sign-in and sign-up components

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/ui/src/components/sign-up/sign-up.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+              <Card.Description>Welcome! Please fill in the details to get started.</Card.Description>
+            </Card.Header>
+            <Card.Body>
+              <Connection.Root>
+                <Common.Connection
+                  name='google'
+                  asChild
+                >
+                  <Connection.Button>
+                    <Common.Icon className='size-4' />
```
[the remark this answers is not in the document]
Comment (alexcarpenter) [reply packages/ui/src/components/sign-up/sign-up.tsx:27]: Yeah, this was just temporary. Will use the icon component you merged in 👍🏼

File: packages/ui/src/components/sign-in/sign-in.tsx
Comment (joe-bell) [inline packages/ui/src/components/sign-in/sign-in.tsx:1]: Could this just be `components/sign-in/sign-in`?

File: packages/ui/src/components/sign-up/sign-up.tsx
Comment (joe-bell) [inline packages/ui/src/components/sign-up/sign-up.tsx:1]: Could this just be `components/sign-up/sign-up`?

File: packages/ui/src/primitives/icon.tsx [hunk 2]
```diff
@@ -14,7 +12,7 @@ function createIcon({ displayName, viewBox, path }: { displayName: string; viewB
         viewBox={viewBox}
         fill='none'
         xmlns='http://www.w3.org/2000/svg'
-        className={cn(iconClassNames, className)}
+        className={cn('size-[1em]', className)}
```
Comment (joe-bell) [inline packages/ui/src/primitives/icon.tsx:15]: Good shout

File: packages/ui/package.json [hunk 3]
```diff
@@ -9,24 +9,33 @@
...
-      ]
+    "./*": {
+      "import": {
+        "types": "./dist/components/*.d.mts",
+        "default": "./dist/components/*.mjs"
+      },
+      "require": {
+        "types": "./dist/components/*.d.ts",
+        "default": "./dist/components/*.js"
+      }
```
Comment (joe-bell) [inline packages/ui/package.json:20]: Ah yes this makes a lot more sense 👍

File: packages/ui/package.json [hunk 4]
```diff
@@ -9,24 +9,33 @@
...
     }
   },
+  "main": "index.js",
```
Comment (joe-bell) [inline packages/ui/package.json:23]: Don't think you need this?

File: packages/ui/package.json [hunk 5]
```diff
@@ -9,24 +9,33 @@
...
+  "files": [
+    "dist"
+  ],
```
Comment (joe-bell) [inline packages/ui/package.json:26]: or this

**Review by joe-bell — APPROVED** [review]

File: packages/ui/src/components/sign-in/sign-in.tsx
Comment (joe-bell) [reply packages/ui/src/components/sign-in/sign-in.tsx:1]: Hmm but couldn't this be `/components/sign-in.tsx` or `/components/sign-in/index.tsx` (same as previous) to achieve the same result? 🤔

**Pull request merged**